client/state/selectors/test/get-current-plan-purchase-id.js

```diff
@@ -0,0 +1,107 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentPlanPurchaseId } from '..';
+
+describe( 'getCurrentPlanPurchaseId()', () => {
+	it( 'should return null if the site is unknown', () => {
+		const state = {
+			sites: {
+				plans: {
+					456: {
+						data: [
+							{
+								currentPlan: true,
+								id: 98765,
+							},
+						],
+					},
+				},
+				items: {
+					456: {},
+				},
+			},
+		};
+
+		expect( getCurrentPlanPurchaseId( state ) ).to.be.null;
+		expect( getCurrentPlanPurchaseId( state, 123 ) ).to.be.null;
+	} );
+
+	it( 'should return null if the current purchase ID is unknown', () => {
+		const siteId = 123;
+		const state = {
+			sites: {
+				plans: {
+					[ siteId ]: {
+						data: [
+							{
+								currentPlan: false,
+								id: 98765,
+							},
+							{
+								currentPlan: true,
+							},
+						],
+					},
+				},
+				items: {
+					[ siteId ]: {},
+				},
+			},
+		};
+
+		expect( getCurrentPlanPurchaseId( state, siteId ) ).to.be.null;
+	} );
+
+	it( 'should return null if there is no plans data, but there is site plan data', () => {
+		const siteId = 123;
+		const state = {
+			sites: {
+				plans: {
+					[ siteId ]: {
+						data: [],
```

Comment:
Test description does not correspond 100% to what this test does - I'm not seeing any site plan data here.